src/main/scala/chisel3/util/BitPat.scala

```diff
@@ -142,6 +143,57 @@ package experimental {
       bs.getWidth
       bs
     }
+
+    /**
+      * Construct a [[BitSet]] matching a range of value
+      *
+      * @param start The smallest matching value
+      * @param length The length of the matching range
+      * @param width The width of the constructed [[BitSet]]. If not given, the returned [[BitSet]] have the width of the maximum possible matching value.
+      * @return A [[BitSet]] matching exactly all inputs in range [start, start + length)
+      */
+    def fromRange(
+      start:  BigInt,
+      length: BigInt,
+      width:  Int = -1
```

Comment:
Rather than using negative values and default values to indicate no `width` argument, it would be better to just overload this function with 1 version taking a width argument and the other one not, eg.
```scala
def fromRange(
   start:  BigInt,
   length: BigInt): BitSet = fromRange(start, length, (start + length - 1).bitLength)


def fromRange(
   start:  BigInt,
   length: BigInt,
   width:  Int
 ): BitSet = ???
```